core/hash_test.go

```diff
@@ -0,0 +1,115 @@
+		},
+	}
+	for _, tt := range tests {
+		t.Run("TestHash", func(t *testing.T) {
+			a, _ := new(felt.Felt).SetString(tt.a)
+			b, _ := new(felt.Felt).SetString(tt.b)
+			ans := new(felt.Felt)
+			want, _ := new(felt.Felt).SetString(tt.want)
+			err := PedersenHash(a, b, ans)
+			if err != nil || !ans.Equal(want) {
+				t.Errorf("TestHash got %s, want %s", ans.Text(16), want.Text(16))
+			}
+		})
+	}
+}
+
+func TestArrayDigest(t *testing.T) {
+	tests := [...]struct {
+		input []string
+		want  string
+	}{
+		// Contract address calculation. See the following links for how the
+		// it is carried out and the result referenced.
+		//
+		//	- https://docs.starknet.io/docs/Contracts/contract-address/
+		//	- https://alpha-goerli.starknet.io/feeder_gateway/get_transaction?transactionHash=0x1b50380d45ebd70876518203f131a12428b2ac1a3a75f1a74241a4abdd614e8
+		{
+			input: []string{
+				// Hex representation of []byte("STARKNET_CONTRACT_ADDRESS").
+				"0x535441524b4e45545f434f4e54524143545f41444452455353",
+				// caller_address.
+				"0x0",
+				// salt.
+				"0x5bebda1b28ba6daa824126577b9fbc984033e8b18360f5e1ef694cb172c7aa5",
+				// contract_hash. See the following for reference https://alpha4.starknet.io/feeder_gateway/get_block?blockHash=0x53e61cb9a53136ecb782e7396f7330e6bb3d069763d866612da3cf93cdf55b5.
+				"0x0439218681f9108b470d2379cf589ef47e60dc5888ee49ec70071671d74ca9c6",
+				// calldata_hash. (here h(0, 0) where h is the Pedersen hash
+				// function).
+				"0x49ee3eba8c1600700ee1b87eb599f16716b0b1022947733551fde4050ca6804",
+			},
+			// contract_address.
+			want: "0x43c6817e70b3fd99a4f120790b2e82c6843df62b573fdadf9e2d677b60ac5eb",
+		},
+		// Transaction hash calculation. See the following for reference.
+		//
+		// 	- https://alpha-mainnet.starknet.io/feeder_gateway/get_transaction?transactionHash=e0a2e45a80bb827967e096bcf58874f6c01c191e0a0530624cba66a508ae75.
+		{
+			input: []string{
+				// Hex representation of []byte("deploy").
+				"0x6465706c6f79",
+				// contract_address.
+				"0x20cfa74ee3564b4cd5435cdace0f9c4d43b939620e4a0bb5076105df0a626c6",
+				// Hex representation of keccak.Digest250([]byte("constructor")).
+				"0x28ffe4ff0f226a9107253e17a904099aa4f63a02a5621de0576e5aa71bc5194",
+				// calldata_hash.
+				"0x7885ba4f628b6cdcd0b5e6282d2a1b17fe7cd4dd536230c5db3eac890528b4d",
+				// chain_id. Hex representation of []byte("SN_MAIN").
+				"0x534e5f4d41494e",
+			},
+			want: "0xe0a2e45a80bb827967e096bcf58874f6c01c191e0a0530624cba66a508ae75",
+		},
+	}
+	for _, test := range tests {
+		data := []*felt.Felt{}
+		for _, item := range test.input {
+			elem, _ := new(felt.Felt).SetString(item)
+			data = append(data, elem)
+		}
+		want, _ := new(felt.Felt).SetString(test.want)
+		got, err := PedersenDigest(data...)
+		if err != nil || got.Cmp(want) != 0 {
+			t.Errorf("PedersenDigest(%x) = %x, want %x", data, got, want)
+		}
+	}
+}
+
+func BenchmarkArrayDigest(b *testing.B) {
```

Comment:
This should also be changed accordingly
```suggestion
func BenchmarkPedersenArrayHash(b *testing.B) {
```